flowey/flowey_lib_common/src/gh_workflow_id.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use flowey::node::prelude::*;
+
+flowey_request! {
+    pub struct Request {
+        pub github_commit_hash: ReadVar<String>,
+        pub repo_path: ReadVar<PathBuf>,
+        pub gh_workflow_id: WriteVar<String>,
+    }
+}
+
+new_simple_flow_node!(struct Node);
+
+impl SimpleFlowNode for Node {
+    type Request = Request;
+
+    fn imports(_ctx: &mut ImportCtx<'_>) {}
+
+    fn process_request(request: Self::Request, ctx: &mut NodeCtx<'_>) -> anyhow::Result<()> {
+        let Request {
+            repo_path,
+            github_commit_hash,
+            gh_workflow_id,
+        } = request;
+
+        let gh_token = ctx.get_gh_context_var(GhContextVar::GITHUB__TOKEN);
+
+        ctx.emit_rust_step("get action id", |ctx| {
+            let gh_workflow_id = gh_workflow_id.claim(ctx);
+            let github_commit_hash = github_commit_hash.claim(ctx);
+            let gh_token = gh_token.claim(ctx);
+            let repo_path = repo_path.claim(ctx);
+
+            |rt| {
+                let github_commit_hash = rt.read(github_commit_hash);
+            let sh = xshell::Shell::new()?;
+            let gh_token = rt.read(gh_token);
+            let repo_path = rt.read(repo_path);
+
+            sh.change_dir(repo_path);
+            // Fetches the CI build workflow id for a given commit hash
+            let get_action_id = |commit: String| {
+            xshell::cmd!(
+                    sh,
+                    "gh run list --commit {commit} -w '[flowey] OpenVMM CI' -s 'completed' -L 1 --json databaseId --jq '.[].databaseId'"
+                )
+                .env("GITHUB_TOKEN", gh_token.clone())
```

Comment:
is this actually necessary? I would expect GitHub Actions to already have this set ambiently?